cocotb/share/lib/fli/FliCbHdl.cpp

```diff
@@ -166,13 +217,17 @@ int FliStartupCbHdl::run_callback()
     product.push_back('\0');
     version.push_back('\0');
 
-
-    // copy in sim_info.product
-    sim_info.argc = 0;
-    sim_info.argv = NULL;
     sim_info.product = &product[0];
     sim_info.version = &version[0];
 
+    std::vector<std::string> argv = get_argv();
+    std::vector<char*> argv_cstr;
```

Comment:
I assume we can't make this `char const*`?

Reply:
sim_info.argv is `char**`, adding const there would require a `const_cast`. But I suppose we should make `sim_info.argv` `const` instead.

Reply:
Let's leave it as is, this is probably motivated by C's bad handling of indirected const.